batch/batch/server/server.py

```diff
@@ -261,7 +320,7 @@ def delete(self):
             batch = batch_id_batch[self.batch_id]
             batch.remove(self)
 
-        self._delete_pod()
+        self._delete_k8s_resources()
         self.set_state('Cancelled')
 
     def is_complete(self):
```

Comment:
Could we make is_complete not have side-effects? Similar to pvc, it's not at all clear to me from is_complete's use that you actually want it to be deleting stuff. It's supposed to check state, and I think based on the results of that state, we should clean up resources.

Reply:
is_complete is a pure function, are you referring to mark_complete?

Reply:
Yep conflated the two when glancing at the diff, meant mark_complete. I think make_complete is a bit complicated, but my comment is just wrong; is_complete is fine!